### DISCUSSION TITLE
How to exclude a package when using the compiler plugin?

### DISCUSSION THREAD

**Discussion by mrclrchtr** [detekt/detekt#6604] [Q&A]
I could not figure out how to exclude packages when using the compiler plugin. Is this currently possible?

It's important for generated sources.

**Comment by BraisGabin** [comment]
🤔 I'm not sure if that's doable right now. . You can exclude by path but we don't have any way to suppress a rule by package.

But I think that it would make a great suppressor. Can you create an issue with this? And, do you want to give it a try?

**Reply by mrclrchtr** [reply]
Thank you for your answer. I am not sure if I have asked the question correctly. It is actually about deactivating a package or a folder for all checks. So e.g. everything in generated sources.
Is this in any way possible with the compiler plug in?

I can open an issue. But I am not sure if I will be able to create a pr in the near future.

**Reply by mrclrchtr** [reply]
@BraisGabin, I have found the problem in the code. The "excludes" is missing in the DetektExtension. I'll open a PR that fixes the problem.

**Reply by mrclrchtr** [reply]
I created https://github.com/detekt/detekt/pull/6611

**Reply by cortinico** [reply]
You can still use the detekt config file to achieve this behavior

**Reply by mrclrchtr** [reply]
Is there a global setting? Or do you mean the individual excludes on the rules?

**Reply by cortinico** [reply]
IIRC You should be able to set set `excludes` at the **ruleset** level. That's not global but close to it

**Reply by mrclrchtr** [reply]
You mean for each rule individually?

What does not work, for example, is
```
formatting:
  active: true
  excludes: [ '**/generated-src/**' ]
  NoConsecutiveBlankLines:
    active: true
    autoCorrect: true
```

**Reply by cortinico** [reply]
Yes correct. If that doesn't work then it's most likely a bug with Detekt

**Reply by mrclrchtr** [reply]
Ok, then it's a bug I think. I created a minimal example here: https://github.com/mrclrchtr/gradle-detekt-compiler-plugin-bug with the exclusion here https://github.com/mrclrchtr/gradle-detekt-compiler-plugin-bug/blob/7da17a8ed8e6e368a7049f9d8640ff2874d4b74e/config.yml#L16 and the failing build here: https://github.com/mrclrchtr/gradle-detekt-compiler-plugin-bug/actions/runs/7832556886/job/21371525166

Should I open an issue?

**Reply by cortinico** [reply]
> Should I open an issue?

Yes please 🙏 and thank you for investigating this

**Reply by mrclrchtr** [reply]
Done: https://github.com/detekt/detekt/issues/6958 

Many thanks for your support!
